src/main/java/com/example/booksstore/service/impl/OrderItemServiceImpl.java

```diff
@@ -0,0 +1,48 @@
+package com.example.booksstore.service.impl;
+
+import com.example.booksstore.dto.orderitem.OrderItemResponseDto;
+import com.example.booksstore.exceptions.NoSuchOrderException;
+import com.example.booksstore.exceptions.NoSuchOrderItemException;
+import com.example.booksstore.mappers.OrderItemMapper;
+import com.example.booksstore.models.OrderItem;
+import com.example.booksstore.repository.OrderItemRepository;
+import com.example.booksstore.repository.OrderRepository;
+import com.example.booksstore.service.OrderItemService;
+import jakarta.transaction.Transactional;
+import java.util.List;
+import java.util.Set;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+
+@Service
+@RequiredArgsConstructor
+public class OrderItemServiceImpl implements OrderItemService {
+    private final OrderItemRepository orderItemRepository;
+    private final OrderItemMapper orderItemMapper;
+    private final OrderRepository orderRepository;
+
+    @Transactional
+    @Override
+    public List<OrderItemResponseDto> getAllOrderItemsByOrderId(Long orderId) {
+        orderRepository.findById(orderId).orElseThrow(() ->
```

Comment:
let's also validate if that order belongs to user who is making the request, to make it more secure and update the error message to something like `... does not exist or does not belong to you`